src/formats/jsonLd.test.ts

```diff
@@ -39,43 +97,27 @@ describe("The Parser", () => {
 
     parser.onQuad(onQuadCallback);
     parser.onComplete(onCompleteCallback);
-    const jsonLd = `
-    {
-      "@id":"https://example.com/some-path#someSubject",
-      "@type":"http://xmlns.com/foaf/0.1/Person",
-      "http://xmlns.com/foaf/0.1/name":"Some name"
-    }
-  `;
-
-    await parser.parse(jsonLd, {
+
+    await parser.parse(jsonLdPersonData, {
```

Comment:
The checks cannot be applied until after all the `onData` callbacks have been triggered which is why the await was originally put here.

The alternative would be to `await` the `onComplete` event but since we already test that behavior using the `stringToArray` function I think it is best to keep the test as-is for now. In particular the current setup ensures that we don't accidentally create regressions in which `onQuad` events occur after the promise resolves (even though consumers shouldn't be relying on that assumption as it is a non-documented part of the API).